runelite-client/src/main/java/net/runelite/client/plugins/chathistory/ChatHistoryPlugin.java

```diff
@@ -68,6 +69,17 @@ protected void shutDown()
 		messageQueue = null;
 	}
 
+	@Subscribe
+	public void onMenuOptionClicked(MenuOptionClicked event)
+	{
+		if (!event.getMenuOption().contains("Clear history"))
+		{
+			return;
+		}
+
+		messageQueue.clear();
```

Comment:
Should probably only clear the messages with types of `PRIVATE_MESSAGE_RECEIVED` `PRIVATE_MESSAGE_SENT` `PRIVATE_MESSAGE_RECEIVED_MOD` since that's what the "clear history" menu option does.